### DISCUSSION TITLE
displaying JTS geometries

### DISCUSSION THREAD

**Discussion by Meibes** [mapsforge/vtm#890] [Q&A] [answered]
Hello,

I have many JTS geometries I want to display on the map and 
[VectorLayer](https://github.com/mapsforge/vtm/blob/87dc04251d3cf9072bef3fee0c8b62c57b0cc646/vtm-jts/src/org/oscim/layers/vector/VectorLayer.java#L57) seems to do the job. When I tested it, I noticed I couldn't iterate over the drawables, nor was I able to remove all of the drawables at once - I need to keep a reference for every drawable to be able to remove it from the VectorLayer. Is this functionality just missing or are there any technical reasons, why these functions are not implemented? As a current workaround I'm working with a [GroupLayer](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/layers/GroupLayer.java), is this a good workaround or do you recommend something else?

greetz

**Comment by devemux86** [comment, marked as answer]
The proper way is to maintain a reference of your layers and drawables.
They are already known when they are created and added.
(or group them somehow with a custom layer)

**Reply by Meibes** [reply]
There seems to be a Problem with VectorLayers in a GroupLayer - if I add VectorLayers to a GroupLayer, only the drawables that have at least one point in the current map extent are drawn - when I pan the map, drawables outside of the original map extent are not drawn. Did I do something wrong? Here are my steps:
- clear GroupLayer
- create & update VectorLayer with drawables
- add VectorLayer to GroupLayer
- update map layers

**Reply by devemux86** [reply]
Better use vector layers directly without such intermediates, like in [examples](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/VectorLayerActivity.java).

GroupLayer is mostly for [simple things](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/SimpleMapActivity.java).
Or create a custom group implementation to work better with your requirements.
